mongo/src/main/scala/io/sphere/mongo/format/DefaultMongoFormats.scala

```diff
@@ -46,15 +46,21 @@ trait DefaultMongoFormats {
   implicit val patternFormat: MongoFormat[Pattern] = new NativeMongoFormat[Pattern]
 
   implicit def optionFormat[@specialized A](implicit f: MongoFormat[A]): MongoFormat[Option[A]] = new MongoFormat[Option[A]] {
+    import scala.collection.JavaConverters._
     override def toMongoValue(a: Option[A]) = a match {
       case Some(aa) => f.toMongoValue(aa)
       case None => MongoNothing
     }
     override def fromMongoValue(any: Any) = {
-      Option(any).map(f.fromMongoValue)
+      Option(any) match {
+        case None => None
+        case Some(dbo: DBObject) if fields.nonEmpty && dbo.keySet().asScala.forall(t ⇒ !fields.contains(t)) => None
```

Comment:
Sorry - what is the `if` supposed to do?
It seems that when the DBObject will fail `f.fromMongoValue`, we instead return a `None`. Is that correct?

If so, why? Shouldn't it fail?

Reply:
If we have an object, and if we find no fields for this object, we return `None`.
The goal is to support https://github.com/sphereio/sphere-scala-libs/pull/75/files#diff-507256a429a967ffb00d7d66bf968820R137
We have an optional embedded object. If we cannot find any field for this object, then we return `None` instead of failing with the error that some required fields are missing.

I did the same for json in https://github.com/sphereio/sphere-scala-libs/pull/12

Reply:
I'll add a test for `MongoFormat[Option[_]]` like https://github.com/sphereio/sphere-scala-libs/pull/12/files#diff-f486508e6406aa0861bf50b93657f494

Reply:
https://github.com/sphereio/sphere-scala-libs/pull/75/commits/8833568e3a6f0fd00114a27c4a1607e320cc7ed4

Reply:
@cneijenhuis tell me if you want further info